developers/weaviate/client-libraries/typescript/v2_v3_migration.md

```diff
@@ -0,0 +1,283 @@
+    }
+  })
+ 
+console.log(client)
+```
+
+</TabItem>
+</Tabs>
+
+Once it has been instantiated, you will notice that the client API is different from `v2`.
+
+## Major changes
+
+From a user's perspective, major changes with the `v3` client include:
+ 
+
+### Better Typescript Support
+
+  - generics, functions and methods
+
+The `v3` client introduces an extensive set of helper classes to interact with Weaviate. These classes are used to provide strong typing, and to make the client more user-friendly such as through IDE autocompletion.
+
+Take a look at the examples below.
+
+import QuickStartCode from '!!raw-loader!/_includes/code/graphql.filters.nearText.generic.py';
+
+<Tabs groupId="languages">
+<TabItem value="create" label="Create a collection">
+
+  <FilteredTextBlock
+    text={PythonCode}
+    startMarker="# START CreateCollectionExample"
+    endMarker="# END CreateCollectionExample"
+    language="py"
+  />
+
+</TabItem>
+<TabItem value="query" label="NearText query">
+
+  <FilteredTextBlock
+    text={QuickStartCode}
+    startMarker="# NearTextExample"
+    endMarker="# END NearTextExample"
+    language="py"
+  />
+
+</TabItem>
+</Tabs>
+
+In both of these examples, you can see how the helper classes and methods abstract away the need for manual JSONs or strings.
+
+### Interaction with collections
+
+Interacting with the `client` object for CRUD and search operations have been replaced with the use of collection objects.
+
+This conveniently removes the need to specify the collection for each operation, and reduces potential for errors.
+
+import ManageDataCode from '!!raw-loader!/_includes/code/howto/manage-data.read.py';
+import ManageDataCodeV3 from '!!raw-loader!/_includes/code/howto/manage-data.read-v3.py';
+
+<Tabs groupId="languages">
+  <TabItem value="py" label="Python (v4)">
+    <FilteredTextBlock
+      text={ManageDataCode}
+      startMarker="# ReadObject START"
+      endMarker="# ReadObject END"
+      language="py"
+    />
+  </TabItem>
+
+  <TabItem value="py3" label="Python (v3)">
+    <FilteredTextBlock
+      text={ManageDataCodeV3}
+      startMarker="# ReadObject START"
+      endMarker="# ReadObject END"
+      language="py"
+    />
+  </TabItem>
+</Tabs>
+
+Note here that the collection object can be re-used throughout the codebase.
+
+### Collection creation from JSON
+
+You can still create a collection from a JSON definition. This may be a useful way to migrate your existing data, for example. You could [fetch an existing definition](../../manage-data/collections.mdx#read-a-single-collection-definition) and then use it to create a new collection.
+
+<FilteredTextBlock
+  text={PythonCode}
+  startMarker="# START CreateCollectionFromJSON"
+  endMarker="# END CreateCollectionFromJSON"
+  language="py"
+/>
+
+### Removal of builder patterns
+
+The builder patterns for constructing queries as been removed, as they could be confusing and potentially lead to invalid queries.
```

Comment:
Consider:
The v2 client syntax used the [builder pattern](link to wikipedia?) to construct queries. The v2 syntax can be confusing and difficult to work with. The v3 client has improved syntax.

Reply:
Updates moved to other branches